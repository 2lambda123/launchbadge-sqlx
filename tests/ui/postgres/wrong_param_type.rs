fn main() {
    let _query = sqlx::query!("select $1::text", 0i32);

    let _query = sqlx::query!("select $1::text", Some(0i32));

    let arg = Some(0i32);
    let _query = sqlx::query!("select $1::text", arg);

    let arg = 0i32;
    let _query = sqlx::query!("select $1::text", arg);
}
