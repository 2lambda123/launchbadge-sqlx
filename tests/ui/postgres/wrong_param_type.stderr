error[E0308]: mismatched types
 --> $DIR/wrong_param_type.rs:2:50
  |
2 |     let _query = sqlx::query!("select $1::text", 0i32);
  |                                                  ^^^^
  |                                                  |
  |                                                  expected struct `std::string::String`, found `i32`
  |                                                  help: try using a conversion method: `0i32.to_string()`

error[E0308]: mismatched types
 --> $DIR/wrong_param_type.rs:4:50
  |
4 |     let _query = sqlx::query!("select $1::text", Some(0i32));
  |                                                  ^^^^^^^^^^ expected struct `std::string::String`, found `i32`
  |
  = note: expected enum `std::option::Option<std::string::String>`
             found enum `std::option::Option<i32>`

error[E0308]: mismatched types
 --> $DIR/wrong_param_type.rs:7:50
  |
7 |     let _query = sqlx::query!("select $1::text", arg);
  |                                                  ^^^
  |                                                  |
  |                                                  expected struct `std::string::String`, found `i32`
  |                                                  help: try using a conversion method: `arg.to_string()`

error[E0308]: mismatched types
  --> $DIR/wrong_param_type.rs:10:50
   |
10 |     let _query = sqlx::query!("select $1::text", arg);
   |                                                  ^^^ expected struct `std::string::String`, found `i32`
   |
   = note: expected enum `std::option::Option<std::string::String>`
              found enum `std::option::Option<i32>`

error[E0308]: mismatched types
  --> $DIR/wrong_param_type.rs:11:50
   |
11 |     let _query = sqlx::query!("select $1::text", arg.as_ref());
   |                                                  ^^^^^^^^^^^^ expected struct `std::string::String`, found `i32`
   |
   = note: expected enum `std::option::Option<&std::string::String>`
              found enum `std::option::Option<&i32>`
